cpp/dolfinx/fem/sparsitybuild.cpp

```diff
@@ -17,12 +17,14 @@ using namespace dolfinx::fem;
 //-----------------------------------------------------------------------------
 void sparsitybuild::cells(
     la::SparsityPattern& pattern, std::span<const std::int32_t> cells,
-    std::array<std::reference_wrapper<const DofMap>, 2> dofmaps)
```

Comment:
To me, it feels odd that the same change is not done in `sparsitybuild::interior_facets`. When I had to customize the sparsity pattern builder in `multiphenicsx` I had to change both `sparsitybuild::interior_facets` and `sparsitybuild::cells`.

Reply:
So far, I've only added changes for cell integrals in this PR. When everyone is happy with the approach, I'll add the other integral types; focusing on cell integrals only for the time being just makes dealing with suggested changes a bit more manageable.